### ISSUE TITLE
Can't transfer Virtual Hydlide save ("Small Buffer Size")

### ISSUE THREAD

**Issue by Gladson-MF** [slinga-homebrew/Save-Game-Extractor#11]
So I've been backing up all my Sega Saturn saves and SGEX works like a charm. However, one specific savegame doesn't seem to work: Virtual Hydlide. It's s pretty large file (uses more than half of Saturn's save capacity) and when I try to transfer it, after 5 minutes or so I get a message:

Small buffer size, possible error: 145

![20221023_095628](https://user-images.githubusercontent.com/116497939/197400677-41f533be-ca5b-45ae-92fb-6eb6ebefb1c9.jpg)

By the way, I thought maybe I could just ignore the error but nope, the generated bin file doesn't seem to be valid.

```
$ python3 sgex.py mysave.bin
Save Game Extractor
(github.com/slinga-homebrew/Save-Game-Extractor)

Reed Solomon couldn't decode buffer, too many errors.
<class 'reedsolo.ReedSolomonError'>
```

**Comment by slinga-homebrew** [comment]
Sorry to hear you are having an issue. I can try to repro but being that the save is compressed first it's unlikely to match your compressed save. 

Can you attach mysave.bin? It might help me debug.

**Comment by Gladson-MF** [comment]
> Sorry to hear you are having an issue. I can try to repro but being that the save is compressed first it's unlikely to match your compressed save.
> 
> Can you attach mysave.bin? It might help me debug.

Sure! I went through the whole process 3 times just in case the bin file is generated slightly different each time, so you have more to work with.

[mysave_bin.zip](https://github.com/slinga-homebrew/Save-Game-Extractor/files/9848060/mysave_bin.zip)

**Comment by slinga-homebrew** [comment]
When you copied it three times, it was the same exact save each time? I'm seeing massive differences. If the files are from the same save I might be able to save them by cobbling 256 byte regions together across the saves.

**Comment by Gladson-MF** [comment]
> When you copied it three times, it was the same exact save each time? I'm seeing massive differences. If the files are from the same save I might be able to save them by cobbling 256 byte regions together across the saves.

Yes, exact same save file. I waited until it finished transmitting (by showing the error message), then closed minimodem, started it again and then pressed C again to start another transfer. Did it three times.

**Comment by Gladson-MF** [comment]
By the way, I'm using minimodem through Cygwin64 on Windows 10, if that makes any difference.

**Comment by slinga-homebrew** [comment]
Would you mind making a few more transfers? I have not been able to correct the save.

**Comment by Gladson-MF** [comment]
> Would you mind making a few more transfers? I have not been able to correct the save.

I tried many, many times, and it just fails every time. I tried with an early game save and it worked so I suppose it's an issue with bigger save files (they seem to get bigger as the game advances).

Do you have any debug ISO that I could burn and send you the results? Or a beta release with a 256 byte copy that I could test, I don't know, anything that could help you debug this issue. Too bad I don't have other means to backup this save so I can send it to you but I'd be glad to help somehow if I can.

**Comment by slinga-homebrew** [comment]
Can you send me the multiple corrupted saves? Assuming they are all of the save exact save I might be able to cobble them together. The three you sent me before were missing too many bytes. 

I can add a feature to dump parts of memory. Will take some time.

**Comment by Gladson-MF** [comment]
> Can you send me the multiple corrupted saves? Assuming they are all of the save exact save I might be able to cobble them together. The three you sent me before were missing too many bytes.
> 
> I can add a feature to dump parts of memory. Will take some time.

Here it is, a package with 20 brand new failed attempts.

[mysave20.zip](https://github.com/slinga-homebrew/Save-Game-Extractor/files/10036361/mysave20.zip)

If there's anything else I can do to help, just let me know!
